src/lightning_app/components/multi_node/pytorch_spawn.py

```diff
@@ -0,0 +1,84 @@
+from typing import Any, Callable, Type
+
+from typing_extensions import Protocol, runtime_checkable
+
+from lightning_app.components.multi_node.base import MultiNode
+from lightning_app.core.work import LightningWork
+from lightning_app.utilities.app_helpers import is_static_method
+from lightning_app.utilities.packaging.cloud_compute import CloudCompute
+from lightning_app.utilities.proxies import WorkRunExecutor
+
+
+@runtime_checkable
+class _PyTorchSpawnProtocol(Protocol):
+    def run(
+        self,
+        world_size: int,
+        node_rank: int,
+        global_rank: int,
+        local_rank: int,
+    ) -> None:
+        pass
+
+
+class _PyTorchSpawnRunExecutor(WorkRunExecutor):
+    def __call__(
+        self,
+        main_address: str,
+        main_port: int,
+        num_nodes: int,
+        node_rank: int,
+    ):
+        import torch
```

Comment:
could we maybe catch the ImportError/ModuleNotFoundError for a nicer error message?

Reply:
Done !